eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/FeeProvider.scala

```diff
@@ -29,54 +30,131 @@ trait FeeProvider {
 
 case object CannotRetrieveFeerates extends RuntimeException("cannot retrieve feerates: channels may be at risk")
 
-// stores fee rate in satoshi/kb (1 kb = 1000 bytes)
-case class FeeratesPerKB(block_1: Long, blocks_2: Long, blocks_6: Long, blocks_12: Long, blocks_36: Long, blocks_72: Long, blocks_144: Long) {
-  require(block_1 > 0 && blocks_2 > 0 && blocks_6 > 0 && blocks_12 > 0 && blocks_36 > 0 && blocks_72 > 0 && blocks_144 > 0, "all feerates must be strictly greater than 0")
+/** Fee rate in satoshi-per-bytes. */
+case class FeeratePerByte(feerate: Satoshi)
 
-  def feePerBlock(target: Int): Long = target match {
+object FeeratePerByte {
+  def apply(feeratePerKw: FeeratePerKw): FeeratePerByte = FeeratePerByte(feeratePerKw.feerate / 250)
```

Comment:
Nit : conversion factors (`250`, `4`, `1000`) seem redundant. Would that be any good to remove one of them and be minimal?

Reply:
True, we only need two of them (the factor 4 between KB and Kw, and the factor 1000 between bytes and kB).

Reply:
Done in https://github.com/ACINQ/eclair/pull/1504/commits/642488a9baa4f93d0ba97f8bcf1c11b1b25c6d33